File: java/minStack.java
class MinStack {
    private ArrayList<Integer> stack;
    private ArrayList<Integer> minStack;
    
    public MinStack(){
        stack = new ArrayList<Integer> ();
        minStack = new ArrayList<Integer> ();
    }
    
    public void push(int x) {
        stack.add(x);
        if(minStack.isEmpty() || x<=this.getMin()){
            minStack.add(x);
        }
    }

    public void pop() throws RuntimeException {
        if(stack.isEmpty()) throw new RuntimeException();
        if(!minStack.isEmpty() && this.top()==this.getMin()){
            minStack.remove(minStack.size()-1);
        }
        stack.remove(stack.size()-1);
    }

    public int top() {
        return stack.get(stack.size()-1);
    }

    public int getMin() {
        return minStack.get(minStack.size()-1);
    }
}

File: python/minstack.py
class MinStack(object):

    def __init__(self):

        """

        initialize your data structure here.

        """

        self.st=[]

        self.minst=[]

        



    def push(self, x):

        """

        :type x: int

        :rtype: nothing

        """

        self.st.append(x)

        if self.minst  :

            if x<=self.minst[-1]:

                self.minst.append(x)

        else:

            self.minst.append(x)

            

    def pop(self):

        """

        :rtype: nothing

        """

        if self.st:

            temp=self.st[-1]

            self.st.pop()

            if self.minst and temp==self.minst[-1]:

                self.minst.pop()

            

        



    def top(self):

        """

        :rtype: int

        """

        if self.st:

            return self.st[-1]

        



    def getMin(self):

        """

        :rtype: int

        """

        if self.minst:

            return self.minst[-1]

        
